repositories/createPost.ts

```diff
@@ -1,11 +1,44 @@
 import * as firebase from "firebase";
-import GameSituation from "../models/GameSituation";
+import { GameType, GameStreetAction } from "../models/GameSituation";
+import PlayingCard from "../models/PlayingCard";
 import { PostTitle, PostBody, PostId } from "../models/Post";
 
 export interface PostData {
   title: PostTitle;
   body: PostBody;
-  gameSituation: GameSituation;
+  gameSituation: PostingGameSituation;
+}
+
+export interface PostingGameSituation {
+  type: GameType;
+  playerLength: number;
+  playerStackSizes: number[];
+  playerCards: ({
+    left: PlayingCard | null;
+    right: PlayingCard | null;
+  } | null)[];
+  heroIndex: number;
+  smallBlindSize: number;
+  antiSize: number;
+  preflop: {
+    actions: GameStreetAction[];
+  };
+  flop?: {
+    communityCards: {
+      left: PlayingCard | null;
+      center: PlayingCard | null;
+      right: PlayingCard | null;
+    };
+    actions: GameStreetAction[];
+  };
+  turn?: {
+    communityCard: PlayingCard | null;
+    actions: GameStreetAction[];
+  };
+  river?: {
+    communityCard: PlayingCard | null;
+    actions: GameStreetAction[];
+  };
```

Comment:
Seems almost complete duplication with `GameSituation` at `models/GameSituation.ts`. Therefore, we'll need to edit two places when we're modifying the form of game situation model. Can you have a good idea to deal with it?

If you don't, let me know and have a pair-programming.

Reply:
I think I can use Partial type